### ISSUE TITLE
dict.get does not show error if key does not exist in Union type

### ISSUE THREAD

**Issue by AlexandrKrivosheev** [microsoft/pyright#1058]
**Describe the bug**
dict.get does not show error if provided key does not exist in Union type

**To Reproduce**
1. Create typed dicts A and B
2. Give variable a type Union[A, B]
3. Try to `.get` not-typed key from variable

**Expected behavior**
Pyright should say that key does not exist in A or B

**Screenshots or Code**
```python
from typing import TypedDict, Union

class A(TypedDict):
	a: str

class B(TypedDict):
	b: str

def get_not_typed_key(ab: Union[A, B]):
	return ab.get("c", None) # "c" key does not exist according to types, but pyright is ok with it
```

**VS Code extension or command-line**
Running pylance vscode ext v2020.9.6.

**Additional context**

**Comment by erictraut** [comment]
The `get` method doesn't require that you pass a defined key. It will return the default value (in your case `None`) if the key isn't present.

Note that if you directly index the dictionary, pyright will flag the error:

```python
def get_not_typed_key(ab: Union[A, B]):
	return ab["c"] # Error reported here
```

**Comment by erictraut** [comment]
I just tried this in mypy, and it does emit an error when `get` is called with an unsupported key, so I guess that's an argument in favor of emitting an error in that case.

**Comment by erictraut** [comment]
I've thought about this more, and I think the current behavior is correct and justified. It is legitimate to call the `get` method with a key that is not in the TypedDict. This call doesn't generate a runtime exception; it returns the default value.

**Comment by AlexandrKrivosheev** [comment]
How can union key be used and type checked then? 🤔 This code has error according to pyright:
```python

def get_typed_key(ab: Union[A, B]):
        if "a" in ab:
	    return ab["a"] # pyright complains that key may not exist

```

**Comment by erictraut** [comment]
That doesn't work as you are expecting because Pyright doesn't currently do type narrowing based on TypedDict key values. The `if "a" in ab` check doesn't narrow the type of `ab` from `Union[A, B]` to `A` in this case. It should be possible to add that feature in cases where TypedDicts in the union are "total" (i.e. all keys are known). I'll consider adding that feature.

**Comment by AlexandrKrivosheev** [comment]
Should I make a feature request to keep track of it? 😄

**Comment by erictraut** [comment]
Thanks for offering. I just created a new feature request to track it. I'm going to close this issue.
